sql/src/main/java/io/crate/rest/action/RestResultSetReceiver.java

```
@@ -105,6 +105,7 @@ public void allFinished(boolean interrupted) {
             super.allFinished(interrupted);
         } catch (Throwable e) {
             LOGGER.error(e);
+            fail(e);
```

We call `super.fail(e)` directly because otherwise we will try to send a response through the channel that is most probably still eroneous or closed. We would also end up logging the same exception again and call `rowAccounting.close()` twice/

---

Thought about that as well. I think the worst thing is that we swallow the original exception. Pushing a fixup.

---

but the previous fail() is not happening on a channel error

---

yes but we can also have a channel fail there :)